test/eventlistener_test.go

```diff
@@ -275,6 +276,11 @@ func TestEventListenerCreate(t *testing.T) {
 		if err != nil {
 			t.Fatalf("Error getting ResourceTemplate: %s: %s", wantPr.Name, err)
 		}
+		if gotPr.Labels[eventIdLabel] == "" {
+			t.Fatalf("Instantiated ResourceTemplate missing EventId")
+		} else {
+			delete(gotPr.Labels, eventIdLabel)
```

Comment:
This seems strange. I think it would make more sense if the expected resources had this label already since there is already a below deep comparison. I think comparing the whole objects rather than just the labels is probably good too.

Reply:
We can not know the expected value for the label so this checks if the label has a value and the rest of the response matches to the expected.

Reply:
Right, I forgot this was an e2e where we couldn't seed the generator.